File: typescript/recorder.ts
import Servirtium, { IServirtium } from '@servirtium/recorder';
import { exec } from 'child_process';

let servirtium: IServirtium;
const baseUrl = 'https://svn.apache.org';

servirtium = new Servirtium(baseUrl);

servirtium.setCallerRequestHeaderReplacements({
    "user-agent: (.*)": "user-agent: Servirtium-Agent"
});
servirtium.setRecordResponseHeaderReplacements({
    "set-cookie: (.*)": "set-cookie: MASKED",
    "date: (.*)": "date: Sun, 09 Aug 2020 18:42:45 GMT",
    "for_testing: (.*)": "for_testing: SERVIRTIUM-REDACTED"
});
servirtium.setRecordResponseHeadersRemoval([
    "x-github-request-id", "expires", "age", "x-timer", "x-cache",
    "x-cache-hits", "x-served-by", "x-fastly-request-id", "x-origin-cache"
]);

// Set the test name for recording
servirtium.setTestName("ExampleSubversionCheckoutRecording");

servirtium.startRecord(() => {
    console.log('Recording started');

    // Wait for 5 seconds before performing the SVN checkout command
    setTimeout(() => {
        console.log('Svn command issued as sub process');
        exec(`svn co http://localhost:61417/repos/asf/synapse/tags/3.0.0/modules/distribution/src/main/conf`, (error, stdout, stderr) => {
            if (error) {
                console.error(`exec error: ${error}`);
                return;
            }
            console.log(`stdout: ${stdout}`);
            console.error(`stderr: ${stderr}`);

            // Ensure the Servirtium proxy is closed and the record is written when done
            servirtium.writeRecord().then(() => {
                console.log('Recording written');
                servirtium.endRecord(() => {

                    console.log('Recording ended');
                    process.exit();
                });
            });
        });
    }, 5000); // 5 seconds delay
});

File: java/src/main/java/Recorder.java
import org.http4k.core.HttpHandler;
import org.http4k.core.Method;
import org.http4k.core.Request;
import org.http4k.core.Response;
import org.http4k.core.Status;
import org.http4k.server.SunHttp;
import org.http4k.servirtium.InteractionStorage;
import org.http4k.servirtium.ServirtiumServer;
import java.io.File;
import java.io.IOException;
import java.util.concurrent.TimeUnit;

public class Recorder {
    public static void main(String[] args) throws IOException {
        String baseUrl = "https://svn.apache.org";
        String interactionFile = "ExampleSubversionCheckoutRecording.md";

        HttpHandler handler = request -> {
            System.out.println("Request: " + request);
            return Response.create().status(Status.OK);
        };

        ServirtiumServer server = ServirtiumServer.Recording(
            baseUrl,
            SunHttp(61417),
            InteractionStorage.Disk(new File(interactionFile)),
            handler
        );

        server.start();
        System.out.println("Recording started");

        try {
            TimeUnit.SECONDS.sleep(5);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

        ProcessBuilder pb = new ProcessBuilder("svn", "co", "http://localhost:61417/repos/asf/synapse/tags/3.0.0/modules/distribution/src/main/conf");
        pb.inheritIO();
        try {
            Process process = pb.start();
            int exitCode = process.waitFor();
            if (exitCode != 0) {
                System.err.println("SVN command failed with exit code " + exitCode);
            }
        } catch (IOException | InterruptedException e) {
            e.printStackTrace();
        }

        server.stop();
        System.out.println("Recording ended");
    }
}
